src/Runner.Worker/JobExtension.cs

```diff
@@ -404,6 +405,18 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                                                                           data: (object)jobHookData));
                     }
 
+                    // Register custom image creation post-job step if the "snapshot" token is present in the message.
```

Comment:
Can you confirm the order? I would imagine the post steps run in the following order:
- Snapshot
- Completed hook
- Job cleanup

Reply:
@TingluoHuang pointed out that the order doesn't seem correct, so I've addressed that concern here: https://github.com/actions/runner/pull/3135/files#r1486509000